src/main/java/io/pravega/connectors/flink/FlinkPravegaInputFormat.java

```diff
@@ -137,7 +137,7 @@ public BaseStatistics getStatistics(BaseStatistics cachedStatistics) throws IOEx
                 for (Iterator<SegmentInfo> segmentInfos = batchClient.listSegments(new StreamImpl(scopeName, stream)); segmentInfos.hasNext(); ) {
                     SegmentInfo segmentInfo = segmentInfos.next();
                     Segment segment = segmentInfo.getSegment();
-                    splits.add(new PravegaInputSplit(splits.size(), segment, 0, segmentInfo.getLength()));
+                    splits.add(new PravegaInputSplit(splits.size(), segment, 0, segmentInfo.getWriteOffset()));
```

Comment:
The constructor of `PravegaInputSplit` is `(..., startOffset, endOffset)`, and the range is inclusive, so this should read:
```
splits.add(new PravegaInputSplit(splits.size(), segment, segmentInfo.getReadOffset(), segmentInfo.getWriteOffset() - 1));
```

@tkaitchuck you agree?